### PR TITLE
Enabled media in ckeditor

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | https://github.com/massiveart/husky/pull/658 , based on #2402 and #2404
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR introduces the tag `sulu:media` and the ui to manage this in the ckeditor. This will not enable image in editor only media-downloads will be included.

Additional content:

* remove `RemoveInternalLink` button from toolbar (now in the overlay)
* style of media-selection-overlay

#### Example Usage

~~~html
<html>
    <body>
        <sulu:media href="1" title="test-title" />
    </body>
</html>
~~~

#### To Do

- [x] Media Tag
- [x] CKEditor plugin for media-selection
  - [x]  upload
  - [x]  style
  - [x]  title
  - [x] remove
  - [x] multiple upload but first taken
- [x] improve css handling for plugins
- [x] translations

### CODE REVIEW

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/ckeditor/internal-link.js [hunk 1]
```diff
@@ -44,6 +44,14 @@ define([
             }
 
             editor.fire('change');
+        },
+        remove = function(editor, selection) {
```
Comment (marcelmoosbrugger) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/ckeditor/internal-link.js]: Space before function

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/ckeditor-internal-link/main.js [hunk 2]
```diff
@@ -110,6 +113,30 @@ define(['underscore', 'text!./form.html'], function(_, formTemplate) {
...
+                    text: this.translations.remove,
+                    align: 'center',
+                    classes: 'just-text',
+                    callback: function() {
+                        this.options.removeCallback();
+                        this.sandbox.emit('husky.overlay.internal-link.close');
+                    }.bind(this)
+                });
+            }
+
```
Comment (marcelmoosbrugger) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/ckeditor-internal-link/main.js]: Can move stuff from this method to new methods. Its way too long.

File: src/Sulu/Bundle/MediaBundle/Markup/MediaTag.php [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+     * {@inheritdoc}
+     */
+    public function parseAll(array $attributesByTag, $locale)
+    {
+        $medias = $this->preloadMedias($attributesByTag, $locale);
+
+        $result = [];
+        foreach ($attributesByTag as $tag => $attributes) {
+            if (!array_key_exists($attributes['id'], $medias)) {
+                $result[$tag] = array_key_exists('content', $attributes) ? $attributes['content'] :
```
Comment (marcelmoosbrugger) [inline src/Sulu/Bundle/MediaBundle/Markup/MediaTag.php]: Can we make something like a "createMarkupForAttributes" method and outsource to body of this foreach?

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/ckeditor/media-link.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+                    editor.contextMenu.addListener(function(element) {
+                        if (element.getAscendant('sulu:media', true)) {
+                            return {
+                                mediaLinkItem: CKEDITOR.TRISTATE_OFF,
+                                removeMediaLinkItem: CKEDITOR.TRISTATE_OFF
+                            };
+                        }
+                    });
+                }
+            },
```
Comment (marcelmoosbrugger) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/ckeditor/media-link.js]: This function is very long.

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/ckeditor/media-link.js [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+                editor.addCommand('mediaLinkDialog', {
+                    dialogName: 'mediaLinkDialog',
+                    allowedContent: 'sulu:media[title,removed,!id]',
+                    requiredContent: 'sulu:media[id]',
+                    exec: function() {
+                        var $element = $('<div/>'),
+                            link = getLinkBySelection(editor.getSelection());
+
+                        $('body').append($element);
```
Comment (marcelmoosbrugger) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/ckeditor/media-link.js]: Child components should not be placed outside the parents element.

**Comment by marcelmoosbrugger** [comment]
Issues when trying it out:
- In the media selection: When single select is activated medias should be selected when clicked (also in masonry view)
- in the media selection: The scrollbars are always visible even when the content does not overflow

File: src/Sulu/Bundle/MediaBundle/Markup/MediaTag.php [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+                )
+            )
+        );
+
+        $fieldDescriptors = $this->fieldDescriptorFactory->getFieldDescriptorForClass(
+            Media::class,
+            ['locale' => $locale]
+        );
+
+        $listBuilder = $this->listBuilderFactory->create(self::$entityName);
```
Comment (marcelmoosbrugger) [inline src/Sulu/Bundle/MediaBundle/Markup/MediaTag.php]: Is it really neccessary to use the list builder here?

**Comment by danrot** [comment]
Replaced by #2621

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/ckeditor/internal-link.js (hunk 1 above)
Comment (marcelmoosbrugger) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/ckeditor/internal-link.js]: fixed in my own branch

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/ckeditor-internal-link/main.js (hunk 2 above)
Comment (marcelmoosbrugger) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/ckeditor-internal-link/main.js]: fixed in my own branch

File: src/Sulu/Bundle/MediaBundle/Markup/MediaTag.php (hunk 3 above)
Comment (marcelmoosbrugger) [inline src/Sulu/Bundle/MediaBundle/Markup/MediaTag.php]: fixed in my own branch

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/ckeditor/media-link.js (hunk 4 above)
Comment (marcelmoosbrugger) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/ckeditor/media-link.js]: fixed in my own branch

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/ckeditor/media-link.js (hunk 5 above)
Comment (marcelmoosbrugger) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/ckeditor/media-link.js]: fixed in my own branch

File: src/Sulu/Bundle/MediaBundle/Markup/MediaTag.php (hunk 6 above)
Comment (marcelmoosbrugger) [inline src/Sulu/Bundle/MediaBundle/Markup/MediaTag.php]: fixed in my own branch

**Pull request closed without merging**